functions/Invoke-DbaDbShrink.ps1

```diff
@@ -0,0 +1,334 @@
+            https://dbatools.io/Invoke-DbaDatabaseShrink
+
+        .EXAMPLE
+            Invoke-DbaDatabaseShrink -SqlInstance sql2016 -Database Northwind,pubs,Adventureworks2014
+
+            Shrinks Northwind, pubs and Adventureworks2014 to have as little free space as possible.
+
+        .EXAMPLE
+            Invoke-DbaDatabaseShrink -SqlInstance sql2014 -Database AdventureWorks2014 -PercentFreeSpace 50
+
+            Shrinks AdventureWorks2014 to have 50% free space. So let's say AdventureWorks2014 was 1GB and it's using 100MB space. The database free space would be reduced to 50MB.
+
+        .EXAMPLE
+            Invoke-DbaDatabaseShrink -SqlInstance sql2014 -Database AdventureWorks2014 -PercentFreeSpace 50 -FileType Data -StepSizeMB 25
+
+            Shrinks AdventureWorks2014 to have 50% free space, runs shrinks in 25MB chunks for improved performance.
+
+        .EXAMPLE
+            Invoke-DbaDatabaseShrink -SqlInstance sql2012 -AllUserDatabases
+
+            Shrinks all databases on SQL2012 (not ideal for production)
+
+    #>
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Low')]
+    param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [Alias("Credential")]
+        [PSCredential]$SqlCredential,
+        [Alias("Databases")]
+        [object[]]$Database,
+        [object[]]$ExcludeDatabase,
+        [switch]$AllUserDatabases,
+        [ValidateRange(0, 99)]
+        [int]$PercentFreeSpace = 0,
+        [ValidateSet('Default', 'EmptyFile', 'NoTruncate', 'TruncateOnly')]
+        [string]$ShrinkMethod = "Default",
+        [ValidateSet('All', 'Data', 'Log')]
+        [string]$FileType = "All",
+        [int]$StepSizeMB,
+        [int]$StatementTimeout = 0,
+        [switch]$LogsOnly,
+        [switch]$ExcludeIndexStats,
+        [switch]$ExcludeUpdateUsage,
+        [Alias('Silent')]
+        [switch]$EnableException
+    )
+
+    begin {
+        if ($LogsOnly) {
+            Test-DbaDeprecation -DeprecatedOn "1.0.0" -Parameter "LogsOnly"
+            $FileType = 'Log'
+        }
+
+        $StepSizeKB = ($stepSizeMB * 1024)
+
+        $StatementTimeoutSeconds = $StatementTimeout * 60
+
+        $sql = "SELECT
+                  avg(avg_fragmentation_in_percent) as [avg_fragmentation_in_percent]
+                , max(avg_fragmentation_in_percent) as [max_fragmentation_in_percent]
+                FROM sys.dm_db_index_physical_stats (DB_ID(), NULL, NULL, NULL, NULL) AS indexstats
+                WHERE indexstats.avg_fragmentation_in_percent > 0 AND indexstats.page_count > 100
+                GROUP BY indexstats.database_id"
+    }
+
+    process {
+        if (!$Database -and !$ExcludeDatabase -and !$AllUserDatabases) {
+            Stop-Function -Message "You must specify databases to execute against using either -Database, -Exclude or -AllUserDatabases" -Continue
+        }
```

Comment:
This is broken code:
- Drop the `-Continue` parameter
- Add a `return` on the next line
- Move to `begin` block
- add the following line as the first line in process instead:

```powershell
if (Test-FunctionInterrupt) { return }
```

Otherwise it can be a danger to the calling function